### PR TITLE
NEW: AUTH (google SSO)

### LINKED ISSUE #13: Authentication using Google SSO
As an EmbedChat developer

**I need to**: Make use of google sso
**So That**: I can sign in or sign-up for EmbeddedChat
**Acceptance Criteria**

- [ ] Create users if they do not exist
- [ ] Change the username using RC token
- [ ] fetch content upon sign-in/sign-up
- [ ] show toast when login fails
- [ ] Read-only for users which are not logged in (use anonymous mode)
- [ ] logout functionality (user friendly logout message)

### LINKED ISSUE #16: Logout user if cookies invalid
### Description:
- Remove the `/me` endpoint hit
- Logout the user when cookies are invalid and the user performs an action

RC authToken doesn't have an expiry, so without checking for the cookies validation in each refresh we need to check if the token is valid when something errors out (like sending a message).

### Expected behavior:
`/me` API call decreases the performance of the app and is not needed

### Actual behavior:
A way to validate requests without worrying about the client side fiddle.

### PR DESCRIPTION
**Pull request by sidmohanty11**
# Authentication

## Acceptance Criteria fulfillment

- [x] Google SSO Auth with RC API
- [x]  Create users if they do not exist
- [x] Change the username using RC token
- [x] fetch content upon sign-in/sign-up
- [x] show toast when the login fails
- [x] Read-only for users who are not logged in
- [x] use anonymous mode
- [x] join the specific channel after successful signup -> [PR](https://github.com/RocketChat/Rocket.Chat/pull/25981)
- [x] logout functionality (user friendly logout message)

Closes #13 
Fixes #16

### CODE REVIEW

**Comment by sidmohanty11** [comment]
## v1

https://user-images.githubusercontent.com/73601258/178520888-b1ddb508-600d-49a2-9d31-8ae54e5c5a69.mp4

**Comment by RonLek** [comment]
@sidmohanty11 have we shifted the "Sign in with Google" button within the message box (if the user isn't logged in) as we discussed? Also the emoji button within the message box shouldn't display if the user isn't signed in.

File: playground/.env[example] [hunk 1]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_RC_HOST=
+REACT_APP_RC_ROOM_ID=
+REACT_APP_GOOGLE_CLIENT_ID=
```
Comment (RonLek) [inline playground/.env[example]:3]: Leave a new line.

**Review by RonLek — CHANGES_REQUESTED** [review]

**Comment by sidmohanty11** [comment]
removed commit (in full) - removed `/me` call and logout user if cookies invalid

when I used this commit message, it said /me isn't a directory and just kept removed as the commit message :(

**Comment by sidmohanty11** [comment]
Done @RonLek sir!

**Comment by sidmohanty11** [comment]
Done @RonLek sir!

**Review by RonLek — APPROVED** [review]
This was a big one! Great work @sidmohanty11 . LGTM.

**Pull request merged**